### PR TITLE
Use `IGitUICommands` in signatures

### PR DESCRIPTION
**Pull request by RussKie**
Follow up to https://github.com/gitextensions/gitextensions/pull/11269

## Proposed changes

- Replace `GitUICommands` with `IGitUICommands` in signatures.
- Move selected types to GitExtensions.Extensibility to enable the above change

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: src/app/GitExtensions.Extensibility/BrowseArguments.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using GitExtensions.Extensibility.Git;
+
+namespace GitExtensions.Extensibility;
```
Comment (mstv) [inline src/app/GitExtensions.Extensibility/BrowseArguments.cs:3]: The namespace block should be kept in order to help git to get that the file has just been moved.

File: src/app/GitExtensions.Extensibility/Git/RevisionDiffKind.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace GitExtensions.Extensibility.Git;
```
Comment (mstv) [inline src/app/GitExtensions.Extensibility/Git/RevisionDiffKind.cs:1]: ditto

**Review by mstv — APPROVED** [review]
👍
just technical comment

**Review by gerhardol — COMMENTED** [review]
+1
OK when mstv approves

File: src/app/GitExtensions.Extensibility/BrowseArguments.cs (hunk 1 above)
Comment (RussKie) [reply src/app/GitExtensions.Extensibility/BrowseArguments.cs]: Keeping it within the namespace doesn't help, the file is still marked as added/deleted.

**Pull request merged**